### PR TITLE
remove reader.batched and use batch_size as argument for train/test/i…

### PR DESCRIPTION
**Pull request by helinwang**
…nfer

our reader definition is func that yield a single item,
it should not be a batch item.
reader.batched creates a func that yield a batch items,
which does not belong to decorator and should not be exposed.

### CODE REVIEW

File: python/paddle/v2/inferencer.py [hunk 1]
```diff
@@ -54,6 +56,12 @@ def default_reader_dict(self):
...
 
 
-def infer(output, parameters, reader, reader_dict=None, field='value'):
+def infer(output,
+          parameters,
+          reader,
+          batch_size,
+          reader_dict=None,
+          field='value'):
+    reader = common.make_batch(reader, batch_size)
```
Comment (typhoonzero) [inline python/paddle/v2/inferencer.py]: 是否需要判断如果bach_size==1，则没必要覆盖reader?

**Comment by hedaoyuan** [comment]
make_batch在train外面是给了用户自定义batch reader的空间。或者可以考虑修改一下train里面的
`for batch_id, data_batch in enumerate(reader()):`，将single item和batch items都支持。

File: python/paddle/v2/inferencer.py (hunk 1 above)
Comment (helinwang) [inline python/paddle/v2/inferencer.py]: 要是batch_size是1，数据是(1,2,3)，仍然需要把它变成[(1,2,3)]，因为train的时候是接受一个batch，而不是一个instance of data。
所以感觉还是需要同样的逻辑处理的。

**Comment by helinwang** [comment]
@hedaoyuan 是说要允许用户使用batch reader吗？现在design doc（https://github.com/PaddlePaddle/Paddle/tree/develop/doc/design/reader）为了尽量的简单，准备就只允许用户返回一条data。

**Comment by wangkuiyi** [comment]
It is invariant that users who write *readers* should just remember that readers yield instances, other than mini-batches.

It is also necessary to allow users customize the batching of instances into mini-batches.

So I suggest that 

- we move `paddle.v2.reader.decorator.batched` into `paddle.batch`

**Pull request closed without merging**